File: code2040-stage1.php
<?php
$request = new stdClass();
$request->token = "1yxife3KZE";
 
$json_data = json_encode($request);
 
$post = file_get_contents('http://challenge.code2040.org/api/getstring',null,stream_context_create(array(
    'http' => array(
        'protocol_version' => 1.1,
        'user_agent'       => 'Diana_Lamaute',
        'method'           => 'POST',
        'header'           => "Content-type: application/json\r\n".
                              "Connection: close\r\n" .
                              "Content-length: " . strlen($json_data) . "\r\n",
        'content'          => $json_data,
    ),
)));
 
if ($post){
	$origStr = json_decode($post)->result;
	$res = strrev($origStr);

	$newRequest = new stdClass();
	$newRequest->token = "1yxife3KZE";
	$newRequest->string = $res;
 
	$json_data = json_encode($newRequest);
 
	$post = file_get_contents('http://challenge.code2040.org/api/validatestring',null,stream_context_create(array(
    'http' => array(
        'protocol_version' => 1.1,
        'user_agent'       => 'Diana_Lamaute',
        'method'           => 'POST',
        'header'           => "Content-type: application/json\r\n".
                              "Connection: close\r\n" .
                              "Content-length: " . strlen($json_data) . "\r\n",
        'content'          => $json_data,
    ),
)));
} 
else echo "POST failed";

File: code2040-stage1.py
#reverse a string

import requests
import json

def strRev(a_string):
	return a_string[::-1]

url = "http://challenge.code2040.org/api/getstring"
data = {"token":"1yxife3KZE"}
headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
r = requests.post(url, data=json.dumps(data), headers=headers)

string = r.json()['result']

result = strRev(string)

url = "http://challenge.code2040.org/api/validatestring"
data = {"token":"1yxife3KZE", "string":result}
headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
r = requests.post(url, data=json.dumps(data), headers=headers)